python/ray/experimental/state.py

```diff
@@ -345,21 +345,55 @@ def task_profiles(self):
         executions of that task. The second element is a list of profiling
         information for tasks where the events have no task ID.
     """
+    task_info = dict()
     event_names = self.redis_client.keys("event_log*")
-    results = dict()
-    events = []
+    counter = 1
     for i in range(len(event_names)):
       event_list = self.redis_client.lrange(event_names[i], 0, -1)
       for event in event_list:
-        event_dict = json.loads(event.decode("ascii"))
-        task_id = ""
-        for element in event_dict:
-          if "task_id" in element[3]:
-            task_id = element[3]["task_id"]
-        if task_id != "":
-          if task_id not in results:
-            results[task_id] = []
-          results[task_id].append(event_dict)
-        else:
-          events.append(event_dict)
-    return results, events
+        event_dict = json.loads(event)
+        tid = ""
+        wid = ""
+        for event in event_dict:
+          if "task_id" in event[3]:
+            tid = event[3]["task_id"]
+        workers = dict()
```

Comment:
We should probably get rid of this numbering. It won't necessarily be consistent across multiple calls to `task_profiles`, and it should be enough to include the worker ID.